app/controllers/purchases_controller.rb

```diff
@@ -1,109 +1,12 @@
 class PurchasesController < ApplicationController
   def create
-    if purchase_params[:gateway] == 'paypal'
-      cart_id = purchase_params[:cart_id]
+    purchase = Purchase::ProcessPurchaseService.call(purchase_params[:gateway], purchase_params[:cart_id],
+                                                    purchase_params[:user], purchase_params[:address])
```

Comment:
Questão de gosto e convenção do projeto, mas acho essa forma mais legível.
```suggestion
    purchase = Purchase::ProcessPurchaseService.call(
      purchase_params[:gateway], 
      purchase_params[:cart_id],
      purchase_params[:user], 
      purchase_params[:address]
    )
```

Também tem a opção de só passar a hash se o seu service sabe lidar dessa forma.
```suggestion
    purchase = Purchase::ProcessPurchaseService.call(purchase_params)
```